cmd/controller/controller.go

```diff
@@ -66,33 +64,6 @@ type Controller struct {
 	updateStatus  bool // feature flag that enables updating the status subresource.
 }
 
-func unseal(ctx context.Context, sclient v1.SecretsGetter, codecs runtimeserializer.CodecFactory, keyRegistry *KeyRegistry, ssecret *ssv1alpha1.SealedSecret) error {
-	// Important: Be careful not to reveal the namespace/name of
-	// the *decrypted* Secret (or any other detail) in error/log
-	// messages.
-
-	objName := fmt.Sprintf("%s/%s", ssecret.GetObjectMeta().GetNamespace(), ssecret.GetObjectMeta().GetName())
-	log.Printf("Updating %s", objName)
-
-	secret, err := attemptUnseal(ssecret, keyRegistry)
-	if err != nil {
-		// TODO: Add error event
-		return err
-	}
-
-	_, err = sclient.Secrets(ssecret.GetObjectMeta().GetNamespace()).Create(ctx, secret, metav1.CreateOptions{})
-	if err != nil && errors.IsAlreadyExists(err) {
-		_, err = sclient.Secrets(ssecret.GetObjectMeta().GetNamespace()).Update(ctx, secret, metav1.UpdateOptions{})
-	}
-	if err != nil {
-		// TODO: requeue?
-		return err
-	}
-
-	log.Printf("Updated %s", objName)
-	return nil
-}
```

Comment:
For the record, the Controller type has an equivalent method:

- https://github.com/bitnami-labs/sealed-secrets/blob/main/cmd/controller/controller.go#L210

Reply:
Anything I should do here?

Reply:
Nope! Just a reference